shared/actions/chat2/index.js

```diff
@@ -933,6 +958,26 @@ function* loadMoreMessages(state, action) {
         key = Constants.getResolvedPendingConversationIDKey(state)
       }
       break
+    case Chat2Gen.loadNewerMessagesDueToScroll:
+      key = action.payload.conversationIDKey
+      reason = 'scroll forward'
+      break
+    case Chat2Gen.loadMessagesFromSearchHit:
+      key = action.payload.conversationIDKey
+      reason = 'search hit'
+      messageIDControl = {
+        mode: RPCChatTypes.localMessageIDControlMode.centered,
+        num: Constants.numMessagesOnInitialLoad,
+        pivot: action.payload.messageID,
```

Comment:
Would be nice not to have to reload everything if the ID is already in the messageMap. Guess it's not easy to find that out here, though.

Reply:
I just did it like this to make it simpler, thought I could revisit later if it is a problem. In practice, it doesn't seem to be too bad and isn't really noticable.